test/models/DepartureStatusSpec.scala

```diff
@@ -19,113 +19,211 @@ package models
 import base.SpecBase
 import generators.ModelGenerators
 import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
-class DepartureStatusSpec extends SpecBase with ScalaCheckDrivenPropertyChecks with ModelGenerators {
+class DepartureStatusSpec extends SpecBase with ScalaCheckDrivenPropertyChecks with ScalaCheckPropertyChecks with ModelGenerators {
 
-  "Initialized must transition" - {
+  "Initialized must" - {
 
-    "to DepartureSubmitted when receiving a DepartureSubmitted event" in {
-      DepartureStatus.Initialized.transition(MessageReceivedEvent.DepartureSubmitted) mustEqual DepartureStatus.DepartureSubmitted
+    "transition to DepartureSubmitted when receiving a DepartureSubmitted event" in {
+      DepartureStatus.Initialized.transition(MessageReceivedEvent.DepartureSubmitted) mustEqual Right(DepartureStatus.DepartureSubmitted)
     }
 
-    "to MrnAllocated when receiving a MrnAllocated event" in {
-      DepartureStatus.Initialized.transition(MessageReceivedEvent.MrnAllocated) mustEqual DepartureStatus.MrnAllocated
+    "transition to MrnAllocated when receiving a MrnAllocated event" in {
+      DepartureStatus.Initialized.transition(MessageReceivedEvent.MrnAllocated) mustEqual Right(DepartureStatus.MrnAllocated)
     }
 
-    "to DepartureRejected when receiving a DepartureRejected event" in {
-      DepartureStatus.Initialized.transition(MessageReceivedEvent.DepartureRejected) mustEqual DepartureStatus.DepartureRejected
+    "transition to DepartureRejected when receiving a DepartureRejected event" in {
+      DepartureStatus.Initialized.transition(MessageReceivedEvent.DepartureRejected) mustEqual Right(DepartureStatus.DepartureRejected)
+    }
+
+    "return an error message when receiving any other event" in {
+      val validMessages   = Seq(MessageReceivedEvent.DepartureSubmitted, MessageReceivedEvent.MrnAllocated, MessageReceivedEvent.DepartureRejected)
+      val invalidMessages = MessageReceivedEvent.values.diff(validMessages)
+      invalidMessages.foreach {
+        m =>
+          DepartureStatus.Initialized.transition(m).isLeft mustBe true
+      }
     }
   }
 
-  "DepartureSubmitted must transition" - {
-    "to DepartureSubmitted when receiving a DepartureSubmitted event" in {
-      DepartureStatus.DepartureSubmitted.transition(MessageReceivedEvent.DepartureSubmitted) mustEqual DepartureStatus.DepartureSubmitted
+  "DepartureSubmitted must" - {
+    "transitionnto DepartureSubmitted when receiving a DepartureSubmitted event" in {
```

Comment:
Can "transition to" be moved into the top level test case title? "DepartureSubmitted must transition to". And the same change made for all transition test cases.

Reply:
I moved the transition to here on purpose because of the check that the error message is created.